src/transformStateWithClones.js

```diff
@@ -7,7 +7,49 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const clones = [];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties':
+        addProperties(stateClone, extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(stateClone, keysToRemove);
+        break;
+
+      case 'clear':
+        clear(stateClone);
+        break;
```

Comment:
You can use the `removeProperties` function to clear all object properties by passing an array of object keys as an argument✨